frodo.js

```diff
@@ -2,79 +2,75 @@
  * Extras Plugin
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.